lib/jekyll/page.rb

```diff
@@ -87,17 +78,6 @@ def template
       end
     end
 
-    # The generated relative url of this page. e.g. /about.html.
-    #
-    # Returns the String url.
-    def url
-      @url ||= URL.new({
-        :template => template,
-        :placeholders => url_placeholders,
-        :permalink => permalink
-      }).to_s
-    end
-
```

Comment:
Is there a way to avoid this API change? The removal or change of these methods will disrupt the stable API for 3.x.  It might not be possible and if it's not is there a way to encapsulate the old into the new with a deprecation message for 4.x?